tiledb/sm/array_schema/auxiliary.cc

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file   auxiliary.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements auxiliary functions to support the ArraySchema class.
+ *
+ * This file exists solely as an intermediary solution to resolve build issues
+ * with `capi_context_stub` when migrating `store_array_schema` out of
+ * `StorageManagerCanonical`. At present, there is an interdependency chain
+ * with `generic_tile_io` which must be resolved before this function can be
+ * placed into `class ArraySchema`. As such, this file is _intentionally_ left
+ * out of the `array_schema` object library and _must_ remain that way. Once
+ * these build issues are resolved, this file should be removed and replaced by
+ * a member function, `ArraySchema::store`.
```

Comment:
```suggestion
 * out of the `array_schema` object library and _must_ remain that way for now.
```

I'm not sure that, once we rework things, that we'll want a member function. This is for a reasons related to the (open as of this writing) #5026. The definition of an array schema is, by design, free of secret data; it all has to be user-visible. In any case, there's no particular reason to telegraph future rework here.